torchgeo/models/btc.py

```diff
@@ -0,0 +1,240 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+#
+# Based on the original code: https://github.com/blaz-r/BTC-change-detection
+
+"""Be The Change (BTC) change detection model implementation."""
+
+import kornia.augmentation as K
+import segmentation_models_pytorch as smp
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from einops import rearrange
+from torch import Tensor
+from torch.nn.modules import Module
+from torchvision.models import Weights, WeightsEnum
+from torchvision.models.feature_extraction import create_feature_extractor
+from torchvision.models.swin_transformer import SwinTransformer, swin_s, swin_t
+
+
+class BTC(Module):
+    """Be The Change (BTC) change detection model.
+
+    If you use this model in your research, please cite the following paper:
+
+    * https://doi.org/10.1109/TGRS.2025.3585342
+    * https://arxiv.org/abs/2507.03367
+    """
+
+    def __init__(self, backbone: str, classes: int = 1) -> None:
+        """Initialise BTC model.
+
+        Args:
+            backbone: backbone type (either swin_tiny, swin_small or swin_base).
+            classes: number of classes (default is 1).
+        """
+        super().__init__()
+        self.backbone = SwinBackbone(backbone)
+        self.difference = subtraction_fusion
+        self.decoder = smp.decoders.upernet.decoder.UPerNetDecoder(
+            encoder_channels=[
+                0,
+                0,
+                *self.backbone.channels,
+            ],  # pad at the beginning since impl. cuts first two off
+            encoder_depth=4,
+            decoder_channels=512,
+        )
+        # we already have layernorms as part of backbone
+        self.decoder.feature_norms = nn.ModuleList(
+            [nn.Identity() for _ in self.backbone.channels]
+        )
+        self.final_layer = smp.base.SegmentationHead(
+            in_channels=512,
+            out_channels=classes,
+            activation=None,
+            kernel_size=1,
+            upsampling=0,  # avoid here in case of uneven factor
+        )
+        smp.base.model.init.initialize_decoder(self.decoder)
+        smp.base.model.init.initialize_head(self.final_layer)
+        # padding tensors used for compatibility with UPerNet implementation
+        self.upernet_padding = [torch.tensor(0), torch.tensor(0)]
+
+    def forward(self, x: Tensor) -> Tensor:
+        """BTC forward call.
+
+        Extract multi-resolution features, fuse by subtraction, decode with UperNet.
+
+        Args:
+            x: input image tensor (b, t*c, h, w)
```

Comment:
Should also document the return value

Reply:
I added this, and also for the other function that didn't have it.